src/map/utils/charutils.cpp

```diff
@@ -1334,15 +1334,12 @@ namespace charutils
             return 0;
         }
 
-        // don't touch this item, it's being used
-        if (PItem->isSubType(ITEM_LOCKED))
-            return 0;
-
-        if (dynamic_cast<CItemState*>(PChar->PAI->GetCurrentState()))
+        auto PState = PChar->PAI->IsCurrentState<CItemState>() ? static_cast<CItemState*>(PChar->PAI->GetCurrentState()) : nullptr;
```

Comment:
just do auto PState = dynamic_cast<CItemState*>(PChar->PAI->GetCurrentState()) - if the current state is CItemState, you get that, if its not, you get nullptr